### DISCUSSION TITLE
Json schema registration overwrite

### DISCUSSION THREAD

**Discussion by sapirpol** [eclipse-theia/theia#8324] [General]
### Bug Description:

I'm familiar with 4 options to register JSON schemas:
1. Add "$schema" with schema's URL in the file.
2. Configure schema in settings.json.
3. Add the following to `package.json`:
```
"contributes": {
    "jsonValidation": [
      {
        "fileMatch": ".jshintrc",
        "url": "https://json.schemastore.org/jshintrc"
      }
    ]
  }
```
4. Enable using schemas coming from schema store.

I tried to overwrite schemas coming from schema store and only succeded by using the first option.
What is the right order of the options and how can I overwrite schemas coming from schema store?

**Comment by bd82** [comment]
Basically is there a specification / definition on this behavior? 

- What is the precedence order between these four registration options?
- Which of these options **adds** additional schema registration to a file?
- Which of these options **overwrites/replaces** the previous schema registrations to a file?

VSCode does not support the SchemaStore yet, but I wonder how it behaves in regards to the first three options.

Relevant references:
- https://github.com/microsoft/vscode/issues/26289#issuecomment-578768254

**Reply by akosyakov** [reply]
Please file a bug request. We should rather get rid of automatic schemas if we cannot make sure that user schemas win.

**Reply by vince-fugnitto** [reply]
> Please file a bug request. We should rather get rid of automatic schemas if we cannot make sure that user schemas win.

Sorry, I moved it from an issue to a discussion since it was mainly questions regarding schemas.

**Reply by sapirpol** [reply]
What's the definition of user schemas?
Is contributes.jsonValidation considered as a user schema and what's the expected behavior of it?

**Reply by akosyakov** [reply]
I meant that schemas contributed by VS Code extensions should have precedence over schemas from the storage.

**Reply by bd82** [reply]
I think the discussion may need to be consolidated in this feature request issue:
- https://github.com/eclipse-theia/theia/issues/8392
